### PR TITLE
Added the conversation metadata support

### PR DESCRIPTION
**Pull request by jaitens**
### What do you want to achieve?
- Added the conversation metadata support

### PR Checklist

- [x] I have tested it locally and all functionalities are working fine.
- [ ] I have compared it with mocks and all design elements are the same.
- [ ] I have tested it in IE Browser.

### How was the code tested?

- locally

NOTE: Make sure you're comparing your branch with the correct base branch

### CODE REVIEW

File: webplugin/js/app/kommunicate-client.js [hunk 1]
```diff
@@ -87,7 +90,9 @@ Kommunicate.client = {
             GROUP_CREATION_URL: window.kommunicate.IFRAME_OVERRIDES ? window.kommunicate.IFRAME_OVERRIDES.GROUP_CREATION_URL : parent.location.href,
             conversationMetadata:JSON.stringify(conversationDetail.metadata),
         };
-    
+        if (kommunicateSettings?.KM_TEST_PAGE) {
```
Comment (devashishmamgain) [inline webplugin/js/app/kommunicate-client.js:93]: we can't have this code in the chat widget, figure out a way to add it through the website test page

**Review by devashishmamgain — CHANGES_REQUESTED** [review]

File: webplugin/js/app/kommunicate-client.js (hunk 1 above)
Comment (jaitens) [reply webplugin/js/app/kommunicate-client.js:93]: we are passing KM_TEST_PAGE key from website test page in the defaultSetting and here we are capturing it.

Comment (devashishmamgain) [reply webplugin/js/app/kommunicate-client.js:93]: i don't think this is the right way to do, there must be a way to achieve the same from website page javascript itself without touching chat widget code.
if required take help from @anujpanwar01

**Review by devashishmamgain — CHANGES_REQUESTED** [review]

File: webplugin/js/app/kommunicate-client.js (hunk 1 above)
Comment (devashishmamgain) [reply webplugin/js/app/kommunicate-client.js:93]: this is weird, @anujpanwar01  is this not possible to pass the key: value pair as used in standard way

**Comment by jaitens** [comment]
closing this PR. handle from website

**Comment by panwaranuj01** [comment]
@jaitens Give the user access to update the default conversation metadata.
Create one function inside the `Kommunicate.js`.

**Comment by jaitens** [comment]
need to create PR from development so closing this PR

**Pull request closed without merging**